### PR TITLE
 added Gbm

### LINKED ISSUE #1095: [Feature]: <Gradient Boosting Machines (GBM)>
### Feature Name

Adding Gradient Boosting Machines (GBM) Visualizations

### Feature Description

Gradient Boosting Machines algorithm, which builds models sequentially and corrects errors made by previous models. This will include popular variants like XGBoost and LightGBM.

### Motivation

GBM techniques are widely used in machine learning competitions and real-world applications due to their high performance. Adding this feature will provide users with advanced modeling capabilities.

### Implementation Suggestions (Optional)

_No response_

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [X] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [ ] Other

### References (Optional)

_No response_

### PR DESCRIPTION
**Pull request by priyashuu**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #1095

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by priyashuu** [comment]
please review this @ajay-dhangar

**Review by ajay-dhangar — APPROVED** [review]
Done
